kvclient,kvserver: clarify comments around DistSender retries

andreimatei opened the pull request:
Also add a TODO about possible double-evaluation of non-transactional
requests.

Release note: None

andreimatei committed:
kvclient,kvserver: clarify comments around DistSender retries

Also add a TODO about possible double-evaluation of non-transactional
requests.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47580)
<!-- Reviewable:end -->

andreimatei commented:
cc @darinpp

andreimatei committed:
roachpb: rectify lying comment on SendError

The comment on SendError suggested that this error is not ambiguous
about whether or not the request that generated the error might have
been evaluated or not. It might have been the case that the intention
was for SendError to not be ambiguous, but it is. Namely, if there's no
commit in the batch, the DistSender will return SendError on networking
errors regardless of whether they're ambiguous or not.
https://github.com/cockroachdb/cockroach/blob/49de3637bfd7b3278aa48c680d481ddd1166fe99/pkg/kv/kvclient/kvcoord/dist_sender.go#L1683

Release note: None

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47580#-:-M5NvwIGBre_1nJqX2bS:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten and @tbg)



<!-- Sent from Reviewable.io -->

andreimatei commented:
bors r+
